x-pack/plugins/search_connectors/server/services/index.ts

```diff
@@ -0,0 +1,230 @@
+  private soClient: SavedObjectsClientContract;
+  private esClient: ElasticsearchClient;
+  private packagePolicyService: PackagePolicyClient;
+  private agentPolicyService: AgentPolicyServiceInterface;
+
+  constructor(
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient,
+    packagePolicyService: PackagePolicyClient,
+    agentPolicyService: AgentPolicyServiceInterface,
+    logger: Logger
+  ) {
+    this.logger = logger;
+    this.soClient = soClient;
+    this.esClient = esClient;
+    this.packagePolicyService = packagePolicyService;
+    this.agentPolicyService = agentPolicyService;
+  }
+
+  public getNativeConnectors = async (): Promise<ConnectorMetadata[]> => {
+    this.logger.debug(`Fetching all connectors and filtering only to native`);
+    const nativeConnectors: ConnectorMetadata[] = [];
+    const allConnectors = await fetchConnectors(this.esClient);
+    for (const connector of allConnectors) {
+      if (connector.is_native && connector.service_type != null) {
+        if (NATIVE_CONNECTOR_DEFINITIONS[connector.service_type] == null) {
+          this.logger.debug(
+            `Skipping connector ${connector.id}: unsupported service type ${connector.service_type}`
+          );
+          continue;
+        }
+
+        nativeConnectors.push({
+          id: connector.id,
+          name: connector.name,
+          service_type: connector.service_type,
+        });
+      }
+    }
+    return nativeConnectors;
+  };
+
+  public getConnectorPackagePolicies = async (): Promise<PackagePolicyMetadata[]> => {
+    const kuery = `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name:${pkgName}`;
+    this.logger.debug(`Fetching policies with kuery: "${kuery}"`);
+    const policiesIterator = await this.packagePolicyService.fetchAllItems(this.soClient, {
+      perPage: 50,
+      kuery,
+    });
+    const policiesMetadata: PackagePolicyMetadata[] = [];
+    for await (const policyPage of policiesIterator) {
+      for (const policy of policyPage) {
+        for (const input of policy.inputs) {
+          if (input.type === connectorsInputName) {
+            if (input.compiled_input != null) {
+              if (input.compiled_input.service_type == null) {
+                this.logger.debug(`Policy ${policy.id} is missing service_type, skipping`);
+                continue;
+              }
+
+              if (input.compiled_input.connector_id == null) {
+                this.logger.debug(`Policy ${policy.id} is missing connector_id, skipping`);
+                continue;
+              }
+
+              if (input.compiled_input.connector_name == null) {
+                this.logger.debug(`Policy ${policy.id} is missing connector_name`);
+                // No need to skip, that's fine
+              }
+
+              if (policy.supports_agentless !== true) {
+                this.logger.debug(`Policy ${policy.id} does not support agentless, skipping`);
+                continue;
```

Comment:
For some reason this doesn't work - I never get a policy that has `supports_agentless` field.